### PR TITLE
feat(ingest): loosen sqlalchemy dep & support airflow 2.3+

### PR DESCRIPTION
**Pull request by hsheth2**
Closes https://github.com/datahub-project/datahub/issues/4809 and closes https://github.com/datahub-project/datahub/issues/6132 and fixes https://github.com/datahub-project/datahub/issues/6145.

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: .github/workflows/metadata-ingestion.yml [hunk 1]
```diff
@@ -54,14 +54,20 @@ jobs:
...
           [
             "installAirflow1",
             "testIntegration",
             "testIntegrationBatch1",
             "testSlowIntegration",
           ]
+        include:
+          - {
+              python-version: "3.7",
+              extraPythonRequirement: "sqlalchemy==1.3.24",
```
Comment (koconder) [inline .github/workflows/metadata-ingestion.yml:67]: @hsheth2 you need to use `>=` instead of `==` to match the rest of the changes

Comment (hsheth2) [reply .github/workflows/metadata-ingestion.yml:68]: This is in the CI test matrix, where we want to test that we remain compatible with SQLAlchemy 1.3.24.

**Comment by tommy-watts-depop** [comment]
Hey, Is there an ETA for this?

**Comment by hsheth2** [comment]
@tommy-watts-depop Can't give a concrete ETA, but I'm hoping to wrap this up mid-next week

**Comment by Vimpy** [comment]
Hey any ETA to merge?

**Comment by KulykDmytro** [comment]
in meanwhile airflow released 2.4.2 version

**Comment by Vimpy** [comment]
Hey any ETA to merge this?

**Review by koconder — APPROVED** [review]
LGTM

**Comment by koconder** [comment]
@Vimpy needs another reviewer to approve the PR

**Review by tommy-watts-depop — APPROVED** [review]

**Comment by koconder** [comment]
Lint tests failing:
```
ERROR: /home/runner/work/datahub/datahub/metadata-ingestion/src/datahub_provider/_plugin.py Imports are incorrectly sorted and/or formatted.
--- /home/runner/work/datahub/datahub/metadata-ingestion/src/datahub_provider/_plugin.py:before	2022-11-10 07:44:16.998670
+++ /home/runner/work/datahub/datahub/metadata-ingestion/src/datahub_provider/_plugin.py:after	2022-11-10 07:52:03.617106
ERROR: /home/runner/work/datahub/datahub/metadata-ingestion/src/datahub_provider/_lineage_core.py Imports are incorrectly sorted and/or formatted.
```

Also smoke tests failing, @hsheth2 can you clean up your commits?

**Comment by keerthiis** [comment]
@hsheth2  - we see this issue with traitlets too :
 acryl-datahub[redshift] 0.8.45 depends on traitlets<5.2.2; extra == "redshift"
    The user requested (constraint file for Airflow 2.3.2) traitlets==5.2.2.post1

**Review by treff7es — APPROVED** [review]
lgtm

**Comment by hsheth2** [comment]
@keerthiis I'd recommend using the PythonVirtualenvOperator in your airflow setup - your root airflow image should only have `acryl-datahub[airflow]` or `acryl-datahub-airflow-plugin` (depending on if you're using the lineage backend or the plugin), and then the specific tasks would depend on `acryl-datahub[<source-name>]`

I'm planning on adding an example to the docs as well

**Pull request merged**